cpp/oneapi/dal/io/detail/common.hpp

```diff
@@ -0,0 +1,172 @@
+/* file: common.hpp */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+#pragma once
+
+#include "oneapi/dal/array.hpp"
+#include "oneapi/dal/common.hpp"
+#include "oneapi/dal/detail/common.hpp"
+
+namespace oneapi::dal::preview::detail {
+
+template <typename T, typename Allocator>
+class vector {
```

Comment:
vector name has to be changed. need to understand do we really need the own vector implementation

Reply:
Please help me with name. container, array is used. custom_array, custom_vector are so bad. vector_with_alloc so long? It is vector, and we need to have vector in the name. Hence, custom_vector. not_std_vector:)

Reply:
vector_container